x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -275,29 +299,32 @@ export class DataRecognizer {
     return body.hits.total.value > 0;
   }
 
-  async listModules() {
-    const manifestFiles = await this.loadManifestFiles();
-    const ids = manifestFiles.map(({ json }) => json.id).sort((a, b) => a.localeCompare(b)); // sort as json files are read from disk and could be in any order.
+  public async listModules() {
+    const manifestFiles = await this._loadConfigs();
+    manifestFiles.sort((a, b) => a.module.id.localeCompare(b.module.id)); // sort as json files are read from disk and could be in any order.
 
-    const modules = [];
-    for (let i = 0; i < ids.length; i++) {
-      const module = await this.getModule(ids[i]);
-      modules.push(module);
+    const configs = [];
```

Comment:
```suggestion
    const configs: Module[] = [];
```
so you can avoid manual casting in the return statement